### PR TITLE
fix #256: change error code from 403 to 401

### LINKED ISSUE #256: "jwt" authenticator returns 403 instead of 401
**"jwt" authenticator returns 403 instead of 401**

From what I see in the code, "jwt" authenticator return 403 (StatusForbidden) in case of authentication failure. I would expect 401 (StatusUnauthorized), the documentation actually says 401 too.

**Reproducing the bug**

1. Access a protected resource with an expired token

```
func (a *AuthenticatorJWT) Authenticate(r *http.Request, config json.RawMessage, _ pipeline.Rule) (*AuthenticationSession, error) {
...
        if err != nil {
                return nil, helper.ErrForbidden.WithReason(err.Error()).WithTrace(err)
        }
```

**Expected behavior**

401 status is returned instead (e.g. the token is not valid).

**Environment**

- latest/master

### PR DESCRIPTION
**Pull request by ngrigoriev**
Signed-off-by: Grigoriev, Nikolai <ngrigoriev@gmail.com>

#256 

## Proposed changes

- added new field to the tests to check the status code for the errors
- changed the status code to 401

## Checklist

- [x ] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md)
- [x ] I have read the [security policy](../security/policy)
- [ x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation within the code base (if appropriate)
- [ ] I have documented my changes in the
      [developer guide](https://github.com/ory/docs) (if appropriate)

## Further comments

Documentation is already mentioning 401 so no updates are needed

### CODE REVIEW

**Comment by ngrigoriev** [comment]
@aeneasr silly me, I have missed the e2e tests that need to be verified in addition to "go test ..."

File: pipeline/authn/authenticator_jwt_test.go [hunk 1]
```diff
@@ -33,6 +33,8 @@ import (
 
 	"github.com/ory/x/urlx"
 
+	"github.com/ory/herodot"
+	"github.com/ory/oathkeeper/driver/configuration"
```
Comment (aeneasr) [inline pipeline/authn/authenticator_jwt_test.go]:
```suggestion
```

Comment (ngrigoriev) [reply pipeline/authn/authenticator_jwt_test.go]: @aeneasr This is odd...my changes did not include this additional import (I only added "herodot") but github shows as if it was added. This import was there before and it is required 6 times in this module.

```
$ grep "configuration." authenticator_jwt_test.go
        "github.com/ory/oathkeeper/driver/configuration"
        viper.Set(configuration.ViperKeyAuthenticatorJWTJWKSURIs, keys)
                                        viper.Set(configuration.ViperKeyAuthenticatorJWTScopeStrategy, "exact")
                                        viper.Set(configuration.ViperKeyAuthenticatorJWTScopeStrategy, "exact")
                                        viper.Set(configuration.ViperKeyAuthenticatorJWTScopeStrategy, "none")
                                        viper.Set(configuration.ViperKeyAuthenticatorJWTScopeStrategy, "none")
```

Comment (aeneasr) [reply pipeline/authn/authenticator_jwt_test.go]: Yeah my bad - I accidentally messed up the merge!

Comment (ngrigoriev) [reply pipeline/authn/authenticator_jwt_test.go]: Oh, now I see, rebasing now

**Pull request merged**